presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -176,6 +177,12 @@ public ActualProperties visitApply(ApplyNode node, List<ActualProperties> inputP
         }
 
         @Override
+        public ActualProperties visitLateralJoin(LateralJoinNode node, List<ActualProperties> inputProperties)
+        {
+            return inputProperties.get(0); // lateral node input (outer query)
```

Comment:
Aren't subquery properties also important? Doesn't properties also depend on join type. see: `visitJoin` method in this class